lib/openhab/dsl/items/ensure.rb

```diff
@@ -38,23 +38,25 @@ module Item
             # def command(state)
             #   return super(state) unless Thread.current[:openhab_ensure_states]
             #
+            #   formatted_state = format_command(state)
             #   logger.trace do
-            #     "#{name} ensure #{state}, format_command: #{format_command(state)}, current state: #{self.state}"
+            #     "#{name} ensure #{state}, format_command: #{formatted_state}, current state: #{raw_state}"
             #   end
-            #   return if self.state == format_command(state)
+            #   return if raw_state == formatted_state
             #
-            #   super(state)
+            #   super(formatted_state)
             # end
             class_eval <<~RUBY, __FILE__, __LINE__ + 1 # rubocop:disable Style/DocumentDynamicEvalDefinition
               def #{ensured_method}(state)
                 return super(state) unless Thread.current[:openhab_ensure_states]
 
+                formatted_state = format_#{ensured_method}(state)
                 logger.trace do
-                  "\#{name} ensure \#{state}, format_#{ensured_method}: \#{format_#{ensured_method}(state)}, current state: \#{self.state}"
+                  "\#{name} ensure \#{state}, format_#{ensured_method}: \#{formatted_state}, current state: \#{raw_state}"
                 end
-                return if self.state == format_#{ensured_method}(state)
+                return if raw_state == formatted_state
 
-                super(state)
+                super(formatted_state)
```

Comment:
Won't `super` also call `format_command`, which means we need to give it the unformatted `state` ?

Reply:
Yes, it will call it, but will immediately return because formatted_type will already be a Type. So we avoid doing the conversion twice.

Reply:
Brilliant!